shell/platform/linux/fl_key_embedder_responder.h

```diff
@@ -15,10 +15,20 @@
 
 constexpr int kMaxConvertedKeyData = 3;
 
-typedef std::function<void(const FlutterKeyEvent* event,
-                           FlutterKeyEventCallback callback,
-                           void* user_data)>
-    EmbedderSendKeyEvent;
+// The signature of a function that FlKeyEmbedderResponder calls on every key
+// event.
+//
+// The `send_key_event_user_data` is a pointer managed by the object that
+// creates FlKeyEmbedderResponder and opaque to FlKeyEmbedderResponder.
```

Comment:
I'd be tempted to explicitly call out the create function by name (even though it's right below) so it's obvious to readers where the `user_data` is being registered that we're getting back here.
```cc
// The `user_data` is opaque data managed by the object that creates
// FlKeyEmbedderResponder, and is registered along with this callback
// via_fl_key_embedder_responder_new.
```